Assets/Scripts/UI/SlipTimeUI.cs

```diff
@@ -9,41 +8,54 @@ public class SlipTimeUI : MonoBehaviour
     {
         public SlipTimeManager slipTimeManager;
 
-        public Text slipTimeUIText;
-
         private int slipTimeCharges;
 
+        // Fillable images for the sliptime meter and charge bars
+        public Image meter;
+        public Image[] bars;
+
         // Start is called before the first frame update
         private void Start()
         {
+            // Handle image setup
+            meter.type = Image.Type.Filled;
+            meter.fillAmount = 1;
+            foreach (var img in bars)
+            {
+                img.type = Image.Type.Filled;
+                img.fillAmount = 1;
+            }
             slipTimeCharges = slipTimeManager.slipTimeCharges;
-            slipTimeUIText.text = $"Slip Time Charges: {slipTimeCharges}";
         }
 
         // Update is called once per frame
-        private void Update()
+        private void LateUpdate()
         {
-            var slipTimeText = $"Slip Time Charges: {slipTimeManager.slipTimeCharges}";
-            
-            // Leave this code here for now. We may be able to re-use it later once we get a better UI for SlipTime
-            // if we don't have to constantly update the text box.
-            // if (slipTimeCharges != slipTimeManager.slipTimeCharges)
-            // {
-            //     slipTimeCharges = slipTimeManager.slipTimeCharges;
-            //     slipTimeUIText.text = $"Slip Time Charges: {slipTimeCharges}";
-            // }
+            if (slipTimeCharges != slipTimeManager.slipTimeCharges)
+            {
+                slipTimeCharges = slipTimeManager.slipTimeCharges;
+                // Reset bars to unfilled
+                foreach (var img in bars)
+                {
+                    img.fillAmount = 0;
+                }
+                // Fill only the bars we need
+                for(var i = 0; i<slipTimeCharges; i++)
+                {
+                    bars[i].fillAmount = 1;
+                }
+            }
 
+            // Sliptime meter shild get emptied when sliptime is engaged
             if (slipTimeManager.InSlipTime)
             {
-                slipTimeText += $"\nSlip Time Duration: {Convert.ToInt32(slipTimeManager.SlipTimeCounter)}";
+                meter.fillAmount = 0;
             }
 
             if (slipTimeManager.IsChargingSlipTime)
```

Comment:
I think this code conflicts with the zeroing out above. Did you want to change it to an `else if` in order to allow a zero fill of the meter if in SlipTime? Or do we want sliptime to recharge even in sliptime?

Reply:
I think we *do* want sliptime to charge during sliptime. Imagine a situation where you've almost charged up a full charge, but then an attack surprises you and you have to engage sliptime to avoid it. If that wipes out your progress towards getting a charge back, you're now basically down *two* charges and have a major #feelsbadman moment. If we do wind up making the charge duration a lot shorter, maybe that wouldn't be as big of a penalty, but with the current tuning, I don't think it would feel great to lose your charge.

All of that is a largely philosophical point, because this UI is currently just reporting what the `SliptimeManager` is telling it. If we want to change the sliptime mechanics, I think we should make a separate PR for modifying the mechanics in `SliptimeManager` if we decide we want to change this.

Reply:
Let me clarify how SlipTime charging works.
When you use your _first_ charge of SlipTime, you don't start recharging _that_ charge until you're out of SlipTime. Once you start charging a charge, you don't stop charging it until you have the charge back.
So you _can_ charge charges during SlipTime, as long as you're _not_ charging the charge you're currently using.
This mimics the behavior of how abilities that have multiple charges tend to work in games that have these kinds of mechanics.

Reply:
Yupp, and that's the behavior that this UI displays.

Reply:
Thanks for taking the time to explain your thoughts. I think we are in agreement here. I think I was confused because of the if statement:

```c#
if (slipTimeManager.InSlipTime) {...}
```

which would effectively remove the fill bar only to be replaced again in the next if statement (unless it's the first time you use sliptime.

I think I would suggest removing that line and instead using:

```c#
if (slipTimeManager.IsChargingSlipTime)
{
    [same code as before]
}
else if (slipTimeManager.InSlipTime)
{
    meter.fillAmount = 0;
}
```

which would save an extra if check.

Reply:
Lol yeah that's a lot cleaner.

Reply:
Thanks!